[Arm64/Linux] SDK fails to build hello world

@richlander @dotnet/arm64-contrib @Petermarcu 

As noted in various places the 2.1 arm64-linux SDK is failing during the build of Hello World with either an assertion or a segmentation fault

The failures all seem to occur shortly after compiling 
```
Enumerator[InstrumentationKind][Microsoft.CodeAnalysis.Emit.InstrumentationKind]::MoveNext
```

The errors take one of these forms
```
Assert failure(PID 37310 [0x000091be], Thread: 37310 [0x91be]): 
    !CREATE_CHECK_STRING(pMT && pMT->Validate())
    File: src/vm/object.cpp Line: 733
```
```
Assert failure(PID 37463 [0x00009257], Thread: 37463 [0x9257]): 
(dac_cast<TADDR>(m_pMethTab) & MARKED_BIT) == 0
    File: src/vm/object.h Line: 192
```
```
Thread 1 "dotnet" received signal SIGSEGV, Segmentation fault.
0x0000ffffb6d58ad0 in MethodTable::GetNumInterfaces (this=0x0) at 
src/vm/methodtable.h:2128
```
```
Thread 1 "dotnet" received signal SIGSEGV, Segmentation fault.
0x0000ffffb6d58ad0 in MethodTable::GetNumInterfaces (this=0x650044002f006a) at 
src/vm/methodtable.h:2128
```
```
error CS8113: Invalid hash algorithm name: '/home/sdmaclea/dotnet-sdk-latest-linux-arm64-2.1.3xx/hw/obj/Debug/netcoreapp2.1/hw.pdb'
```
```
Thread 1 "dotnet" received signal SIGSEGV, Segmentation fault.
0x0000ffffb6cacf28 in MethodTable::GetFlag (this=0x6200640070002c, flag=MethodTable::enum_flag_HasComponentSize) at 
```

This seems like there is a bad argument passed to the compiler -- the value of PdbChecksumAlgorithm should be something like `SHA256`, not a path to a PDB:

```
error CS8113: Invalid hash algorithm name: '/home/sdmaclea/dotnet-sdk-latest-linux-arm64-2.1.3xx/hw/obj/Debug/netcoreapp2.1/hw.pdb'
```

--- a/src/jit/codegenarmarch.cpp
+++ b/src/jit/codegenarmarch.cpp
@@ -655,6 +655,12 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
             // Deal with the multi register passed struct args.
             GenTreeFieldList* fieldListPtr = source->AsFieldList();
 
+#ifdef _TARGET_ARM64_
+            // Arm64 ABI does not include argument splitting between registers and stack
+            assert(fieldListPtr);
+            assert(fieldListPtr->gtFieldOffset == 0);
+#endif // _TARGET_ARM64_
+
             // Evaluate each of the GT_FIELD_LIST items into their register
             // and store their register into the outgoing argument area
             for (; fieldListPtr != nullptr; fieldListPtr = fieldListPtr->Rest())
@@ -666,11 +672,21 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
                 var_types type = nextArgNode->TypeGet();
                 emitAttr  attr = emitTypeSize(type);
 
+#ifdef _TARGET_ARM64_
+                // Emit store instructions to store the registers produced by the GT_FIELD_LIST into the outgoing
+                // argument area
+                emit->emitIns_S_R(ins_Store(type), attr, reg, varNumOut, argOffsetOut + fieldListPtr->gtFieldOffset);
+
+                // We can't write beyound the outgoing area area
+                assert((argOffsetOut + fieldListPtr->gtFieldOffset + EA_SIZE_IN_BYTES(attr)) <= argOffsetMax);
+#else
+                // TODO-ARM-Bug?  The following code will pack copied structs
                 // Emit store instructions to store the registers produced by the GT_FIELD_LIST into the outgoing
                 // argument area
                 emit->emitIns_S_R(ins_Store(type), attr, reg, varNumOut, argOffsetOut);
                 argOffsetOut += EA_SIZE_IN_BYTES(attr);
                 assert(argOffsetOut <= argOffsetMax); // We can't write beyound the outgoing area area
+#endif // _TARGET_ARM64_
             }
         }
         else // We must have a GT_OBJ or a GT_LCL_VAR

--- a/src/jit/morph.cpp
+++ b/src/jit/morph.cpp
@@ -5497,7 +5497,6 @@ GenTreeFieldList* Compiler::fgMorphLclArgToFieldlist(GenTreeLclVarCommon* lcl)
     LclVarDsc* varDsc = &(lvaTable[lcl->gtLclNum]);
     assert(varDsc->lvPromoted == true);
 
-    unsigned          offset      = 0;
     unsigned          fieldCount  = varDsc->lvFieldCnt;
     GenTreeFieldList* listEntry   = nullptr;
     GenTreeFieldList* newArg      = nullptr;
@@ -5508,12 +5507,12 @@ GenTreeFieldList* Compiler::fgMorphLclArgToFieldlist(GenTreeLclVarCommon* lcl)
     {
         LclVarDsc* fieldVarDsc = &lvaTable[fieldLclNum];
         GenTree*   lclVar      = gtNewLclvNode(fieldLclNum, fieldVarDsc->lvType);
-        listEntry = new (this, GT_FIELD_LIST) GenTreeFieldList(lclVar, offset, fieldVarDsc->lvType, listEntry);
+        listEntry              = new (this, GT_FIELD_LIST)
+            GenTreeFieldList(lclVar, fieldVarDsc->lvFldOffset, fieldVarDsc->lvType, listEntry);
         if (newArg == nullptr)
         {
             newArg = listEntry;
         }
-        offset += TARGET_POINTER_SIZE;
         fieldLclNum++;
     }
     return newArg;

